service/frontend/workflow_handler_test.go

```diff
@@ -1496,6 +1515,176 @@ func (s *workflowHandlerSuite) TestGetHistory() {
 	s.EqualValues(`"random-data"`, history.Events[1].GetWorkflowExecutionStartedEventAttributes().GetSearchAttributes().GetIndexedFields()["TemporalChangeVersion"].GetData())
 }
 
+func (s *workflowHandlerSuite) TestGetHistoryError() {
+	namespaceID := namespace.ID(uuid.New())
+	namespaceName := namespace.Name("test-namespace")
+	firstEventID := int64(100)
+	nextEventID := int64(102)
+	branchToken := []byte{1}
+	we := commonpb.WorkflowExecution{
+		WorkflowId: "wid",
+		RunId:      "rid",
+	}
+	shardID := common.WorkflowIDToHistoryShard(namespaceID.String(), we.WorkflowId, numHistoryShards)
+	req := &historyservice.ReadHistoryBranchRequest{
+		NamespaceId:   namespaceID.String(),
+		ShardId:       shardID,
+		BranchToken:   branchToken,
+		MinEventId:    firstEventID,
+		MaxEventId:    nextEventID,
+		PageSize:      3,
+		NextPageToken: []byte{},
+	}
+	s.mockHistoryClient.EXPECT().ReadHistoryBranch(gomock.Any(), req).Return(nil, serviceerror.NewDataLoss("test"))
+
+	wh := s.getWorkflowHandler(s.newConfig())
+
+	_, _, err := wh.getHistory(
+		context.Background(),
+		metrics.NoopMetricsHandler,
+		namespaceID,
+		namespaceName,
+		we,
+		firstEventID,
+		nextEventID,
+		3,
+		[]byte{},
+		nil,
+		branchToken,
+	)
+	s.Error(err)
+}
+
+func (s *workflowHandlerSuite) TestGetHistoryReverse() {
+	namespaceID := namespace.ID(uuid.New())
+	namespaceName := namespace.Name("test-namespace")
+	nextEventID := int64(103)
+	lastFirstTxnID := int64(1000)
```

Comment:
This name is so confusing. Could you try to investigate what this name means and find a better one?